src/java_bytecode/generate_java_generic_type.cpp

```diff
@@ -32,44 +36,49 @@ symbolt generate_java_generic_typet::operator()(
   INVARIANT(
     pointer_subtype.id()==ID_struct, "Only pointers to classes in java");
 
-  const java_class_typet &java_class=to_java_class_type(pointer_subtype);
-  java_class_typet replacement_type=java_class;
-  replacement_type.components().clear();
-
-  const irep_idt new_tag=build_generic_tag(existing_generic_type, java_class);
+  java_class_typet &replacement_type=to_java_class_type(const_cast<typet &>(pointer_subtype));
```

Comment:
What is going on here? It appears you are casting away const-ness(?!) which means you'd actually end up modifying the components on the original type? I don't think you need to create a `java_class_typet` here since you create one in `generate_class_stub`, it should be sufficent to make a new `componentst` that is a _copy_ of the `pointer_subtype.components()`. Suggest you extend your tests to check that the original struct is still present with the original type for the components.